File: StringCalculator/StringCalculatorTest.cs
using System;
using NUnit.Framework;

namespace StringCalculator
{
    [TestFixture]
    public class StringCalculatorTest

    {
        [Test]
        public void EmptyStringReturnsZero()
        {
            Assert.AreEqual(0, StringCalculator.Add(""));
        }

        [Test]
        public void BasicAddition()
        {
            Assert.AreEqual(3, StringCalculator.Add("1,2"));
        }

        [Test]
        public void GetBasicDelimiter()
        {
            const string input = "//;\n1;2";
            Assert.AreEqual(";", StringCalculator.GetDelimiter(input));
        }

        [Test]
        public void OnePlusTwo()
        {
            var result = StringCalculator.Add("1,2");
            Assert.AreEqual(3, result);
        }

        [Test]
        public void OnePlusTwoPlus3()
        {
            var result = StringCalculator.Add("1,2,3");
            Assert.AreEqual(6, result);
        }

        [Test]
        public void OnePlusTwoWithNewLineNAndComma()
        {
            var result = StringCalculator.Add("1\n2,3");
            Assert.AreEqual(6, result);
        }

        [Test]
        public void OnePlusTwoWithNewLineRAndComma()
        {
            var result = StringCalculator.Add("1\r2,3");
            Assert.AreEqual(6, result);
        }

        [Test]
        public void OnePlusTwoWithNewLinesRandN()
        {
            var result = StringCalculator.Add("1\r2\n3");
            Assert.AreEqual(6, result);
        }

        [Test]
        public void OnePlusOneWithDelimiter()
        {
            var result = StringCalculator.Add("//;\n1;1");
            Assert.AreEqual(2, result);
        }

        [Test]
        public void ExceptionMessageIsCorrectForOneNegative()
        {
            var ex = Assert.Throws<ArgumentException>(() => StringCalculator.Add("-4,1"));
            Assert.That(ex.Message, Is.EqualTo("Negatives are not allowed: -4 "));
        }

        [Test]
        public void ExceptionMessageIsCorrectForThreeNegatives()
        {
            var ex = Assert.Throws<ArgumentException>(() => StringCalculator.Add("1,2,3,1,-6,12,-3,34,-6"));
            Assert.That(ex.Message, Is.EqualTo("Negatives are not allowed: -6 -3 -6 "));
        }

        [Test]
        public void Adds1000()
        {
            var result = StringCalculator.Add("1,2,1000");
            Assert.AreEqual(1003, result);
        }

        [Test]
        public void IgnoresNumbersGreaterThan1000()
        {
            var result = StringCalculator.Add("1,1,1001,1,2345");
            Assert.AreEqual(3, result);
        }

        [Test]
        public void RandomLengthDelimiter()
        {
            const string input = "//[***]\n1***2***3";
            var result = StringCalculator.GetDelimiter(input);
            Assert.AreEqual("***", result);
        }

        [Test]
        public void DefaultDelimiter()
        {
            const string input = "1,2";
            Assert.AreEqual("[,\n\r]", StringCalculator.GetDelimiter(input));
        }
    }
}

File: java/src/com/monktowntech/stringCalculator/StringCalculatorTest.java
package com.monktowntech.stringCalculator;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class StringCalculatorTest {
  StringCalculator sc = new StringCalculator();

  @Test
  void emptyStringReturnsZero() {
    int result = sc.add("");
    assertEquals(0, result);
  }

  @Test
  void basicAddition() {
    assertEquals(3, sc.getSum(new String[]{"1", "1", "1"}));
  }

  @Test
  void getBasicDelimiter() {
    String input = "//;\n1;2";
    assertEquals(";", sc.getDelimiter(input));
  }

  @Test
  void onePlusTwo() {
    int result = sc.add("1,2");
    assertEquals(3, result);
  }

  @Test
  void onePlusTwoPlus3() {
    int result = sc.add("1,2,3");
    assertEquals(6, result);
  }

  @Test
  void onePlusTwoWithNewLineNAndComma() {
    int result = sc.add("1\n2,3");
    assertEquals(6, result);
  }

  @Test
  void onePlusTwoWithNewLineRAndComma() {
    int result = sc.add("1\r2,3");
    assertEquals(6, result);
  }

  @Test
  void onePlusTwoWithNewLinesRandN() {
    int result = sc.add("1\r2\n3");
    assertEquals(6, result);
  }

  @Test
  void onePlusOneWithDelimiter() {
    int result = sc.add("//;\n1;1");
    assertEquals(2, result);
  }

  @Test
  void negativeNumbersThrowException() {
    Assertions.assertThrows(IllegalArgumentException.class, () -> sc.add("4,-1"));
  }

  @Test
  void exceptionMessageIsCorrectForOneNegative() {
    String message = "";
    try {
      sc.add("4,-1");
    } catch (IllegalArgumentException e) {
      message = e.getMessage();
    }
    assertEquals("Negatives not allowed: -1 ", message);
  }

  @Test
  void exceptionMessageIsCorrectForThreeNegatives() {
    String message = "";
    try {
      sc.add("4,-1,5,-6,7,-5");
    } catch (IllegalArgumentException e) {
      message = e.getMessage();
    }
    assertEquals("Negatives not allowed: -1 -6 -5 ", message);
  }

  @Test
  void adds1000() {
    int result = sc.add("1,2,1000");
    assertEquals(1003, result);
  }

  @Test
  void ignoresNumbersGreaterThan1000() {
    int result = sc.add("1,1,1001,1,2345");
    assertEquals(3, result);
  }

  @Test
  void randomLengthDelimiter() {
    String input = "//[***]\n1***2***3";
    String result = sc.getDelimiter(input);
    assertEquals("***", result);
  }

  @Test
  void defaultDelimiter() {
    String input = "1,2";
    assertEquals("[,\n\r]", sc.getDelimiter(input));
  }


}
